src/field.rs

```
@@ -15,12 +15,83 @@ use std::ops::Deref;
+            TIBRVMSG_U8ARRAY => fld.try_decode().map(DecodedField::U8Array),
+            TIBRVMSG_I8 => fld.try_decode().map(DecodedField::I8),
+            TIBRVMSG_I8ARRAY => fld.try_decode().map(DecodedField::I8Array),
+            TIBRVMSG_U16 => fld.try_decode().map(DecodedField::U16),
+            TIBRVMSG_U16ARRAY => fld.try_decode().map(DecodedField::U16Array),
+            TIBRVMSG_I16 => fld.try_decode().map(DecodedField::I16),
+            TIBRVMSG_I16ARRAY => fld.try_decode().map(DecodedField::I16Array),
+            TIBRVMSG_U32 => fld.try_decode().map(DecodedField::U32),
+            TIBRVMSG_U32ARRAY => fld.try_decode().map(DecodedField::U32Array),
+            TIBRVMSG_I32 => fld.try_decode().map(DecodedField::I32),
+            TIBRVMSG_I32ARRAY => fld.try_decode().map(DecodedField::I32Array),
+            TIBRVMSG_U64 => fld.try_decode().map(DecodedField::U64),
+            TIBRVMSG_U64ARRAY => fld.try_decode().map(DecodedField::U64Array),
+            TIBRVMSG_I64 => fld.try_decode().map(DecodedField::I64),
+            TIBRVMSG_I64ARRAY => fld.try_decode().map(DecodedField::I64Array),
+            TIBRVMSG_F32 => fld.try_decode().map(DecodedField::F32),
+            TIBRVMSG_F32ARRAY => fld.try_decode().map(DecodedField::F32Array),
+            TIBRVMSG_F64 => fld.try_decode().map(DecodedField::F64),
+            TIBRVMSG_F64ARRAY => fld.try_decode().map(DecodedField::F64Array),
+            TIBRVMSG_BOOL => fld.try_decode().map(DecodedField::Bool),
+            TIBRVMSG_DATETIME => fld.try_decode().map(DecodedField::DateTime),
+            TIBRVMSG_IPADDR32 => fld.try_decode().map(DecodedField::Ipv4),
+            TIBRVMSG_IPPORT16 => tibrv_try_decode_port(fld).map(DecodedField::IpPort),
+            TIBRVMSG_OPAQUE => unsafe { tibrv_try_decode_opaque::<u8>(fld).map(DecodedField::Opaque) },
+            _ => Err(ErrorKind::FieldTypeError.into()), // FIXME: should be more specific error type
```

I suggest `ErrorKind::UnknownFieldType`, it's probably never going to happen so you're right we're better to throw an error than include an `Unknown` variant or such.

Maybe we want to rename `FieldTypeError` to be more specific then?